optaplanner-core/src/main/java/org/optaplanner/core/impl/domain/solution/descriptor/SolutionDescriptor.java

```diff
@@ -977,14 +991,39 @@ public long getProblemScale(Solution_ solution) {
                 .sum();
     }
 
-    public int countUninitializedVariables(Solution_ solution) {
+    /**
+     * Calculates the number of elements that need to be processed in the Construction Heuristics phase.
+     * The negative value of this is the {@code initScore}. It represents how many Construction Heuristics steps need to
+     * be taken before the solution is fully initialized.
+     *
+     * @param solution never null
+     * @return {@code >= 0}
+     */
+    public int countUninitialized(Solution_ solution) {
```

Comment:
If I am reading this code correctly, it get the number of unassigned values in the first list variable descriptor found. If there is no list variable descriptor, it count the number of uninitialized variables. Can regular planning variables be used with list variables? Can multiple list variables exist (as different fields/on different classes)?

Reply:
> Can regular planning variables be used with list variables?

No, it fails fast if a basic variable is used together with a list variable on the same planning entity.

I don't know what happens if there are multiple entities and one or more of them uses a list variable. Will have to try it to find out.